File: 0071-simplify-path/0071-simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        vector<string> stack;
        stringstream ss(path);
        string temp;

        while (getline(ss, temp, '/')) {
            if (temp == "." || temp.empty() == true) {
                continue;
            }
            else if (temp == "..") {
                if (stack.empty() == false) {
                    stack.pop_back();
                }
            }
            else {
                stack.push_back(temp);
            }
        }

        string ans = "";
        for (string dir : stack) {
            ans += "/" + dir;
        }
        if (ans.empty() == true) {
            return "/";
        }
        else {
            return ans;
        }

    }
};

File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        split_path = path.split('/')
        for char in split_path:
            if char == "" or char == ".":
                continue
            elif char == "..":
                if stack:
                    stack.pop()
            else:
                stack.append(char)
        return "/" + "/".join(stack)